llvm/unittests/CAS/LazyAtomicPointerTest.cpp

```diff
@@ -34,26 +35,29 @@ TEST(LazyAtomicPointer, loadOrGenerate) {
   EXPECT_EQ(Ptr.load(), &Value);
 }
 
+#if (LLVM_ENABLE_THREADS)
 TEST(LazyAtomicPointer, BusyState) {
   int Value = 0;
   LazyAtomicPointer<int> Ptr;
   ThreadPool Threads;
 
-  std::mutex BusyStart, BusyEnd;
-  BusyStart.lock();
-  BusyEnd.lock();
+  std::mutex BusyLock, EndLock;
+  std::condition_variable Busy, End;
   Threads.async([&]() {
     Ptr.loadOrGenerate([&]() {
-      BusyStart.unlock();
-      while (!BusyEnd.try_lock()) {
-        // wait till the lock is unlocked.
-      }
+      // Notify busy state.
+      Busy.notify_all();
+      std::unique_lock<std::mutex> LEnd(EndLock);
+      // Wait for end state.
+      End.wait(LEnd);
+      LEnd.unlock();
```

Comment:
Why explicitly unlock? It goes out of scope.

Reply:
Windows thread will not even return if you don't unlock.